manage_people/views.py

```diff
@@ -316,9 +322,22 @@ def add_users(request):
     for user_id, user_role_id in list(users_to_add.items()):
         # Add the returned (existing_enrollment, person) tuple to the results
         # list
-        enrollment_results.append(
-            add_member_to_course(user_id, int(user_role_id), course_instance_id,
-                                 canvas_course_instance_id))
+        existing, person, error_msg = add_member_to_course(
```

Comment:
Might be better to wrap this in a try/except and get the error message from the except here vs getting it from a tuple value. This way you could just raise in the `add_member_to_course` function and avoid further processing. Open to discuss

Reply:
I created a class to cover custom exceptions and placed it towards the top of the views file. Wasn't totally sure where to place it but I use to raise errors.
```
class EnrollmentError(Exception):
    """Custom exception raised when a user enrollment fails."""
    def __init__(self, message, user_id=None):
        self.message = message
        self.user_id = user_id
        super().__init__(self.message)
```